WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person Payments/CardPresentPaymentsOnboardingUseCase.swift

```diff
@@ -91,7 +110,12 @@ final class CardPresentPaymentsOnboardingUseCase: CardPresentPaymentsOnboardingU
         }
 
         let paymentGatewayAccountsAction = CardPresentPaymentAction.loadAccounts(siteID: siteID) { [weak self] result in
-            self?.updateState()
+            guard let self = self else {
+                return
+            }
+
+            self.updateState()
+            CardPresentPaymentOnboardingStateCache.shared.update(self.state)
```

Comment:
`updateState` gets called from `selectPlugin` as well... why don't we update the cache there too?

Reply:
Good catch! Done in fb904a6fd09a944f63b9fdd57671391da6f96c0e